forui/lib/src/widgets/header/nested_header.dart

```diff
@@ -0,0 +1,166 @@
+part of 'header.dart';
+
+/// A nested header.
+///
+/// A nested header contains the page's title and actions (including pop navigation).
+/// It is typically used on pages that are not at the root of the navigation stack.
+///
+/// See:
+/// * https://forui.dev/docs/header for working examples.
+/// * [FNestedHeaderStyle] for customizing a header's appearance.
```

Comment:
Private classes don't need documentation.
```suggestion
```

Reply:
I left it there because I was thinking if the classes should be private in the first place. I see the header as a sugar syntax for accessing the actual root and nested header.

It could make sense to leave the comments in for now since there's a good chance we will expose these classes publicly.